### PR TITLE
TRAIN-272: creating login page

### PR DESCRIPTION
**Pull request by Stanley-Maraba**

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import Header from './header/Header'
+import ManageLoginData from './ManageLoginData';
+import {Route, Router, Routes} from "react-router-dom";
+import HomePage from './HomePage';
+import ManageRegisterData from "./ManageRegisterData";
+import PageNotFound from './PageNotFound'
```
Comment (colsen-neslo) [inline src/components/App.jsx]: Please follow the following standard when defining react imports.
https://github.com/NesloTechnologies/eventhub-ui-kit/blob/develop/CONTRIBUTING.md
```JavaScript
// vendor/3rd party imports
import React from 'react';
import Accordion from '@material-ui/core/components';
import { string, bool, arrayOf } from 'prop-types';

// internal neslo imports
import { Button } from '@neslotech/eventhub-ui-kit';

// helpers, actions, utilities, etc.
import { requestUsernameValidation } from '../../actions/auth.actions';
import { convertDateToTime } from '../../helper/date.helper';

// components defined in project
import UserProfile from '../user-profile/UserProfile';

// stylesheets
import './get-started.scss'; 
```

File: src/components/App.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+function App() {
```
Comment (colsen-neslo) [inline src/components/App.jsx]: Use an arrow function.

File: src/components/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import "./login.css";
+import {NavLink} from "react-router-dom";
+
+function Login(props) {
+
+    return (
+        <>
+            <h1>Login</h1>
+
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: Remove the un-necessary blank lines between elements in this file please.

File: src/components/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import "./login.css";
+import {NavLink} from "react-router-dom";
+
+function Login(props) {
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: Use arrow function.

File: src/components/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import "./login.css";
+import {NavLink} from "react-router-dom";
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: Fix import ordering.

File: src/components/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+function Login(props) {
+
+    return (
+        <>
+            <h1>Login</h1>
+
+            <div id="login">
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: Please make use of semantically correct HTML. Don't use div where div is not required or semantically correct.
```suggestion
            <section id="login">
```
Or
```suggestion
            <article id="login">
```

File: src/components/ManageLoginData.jsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React,{useState} from "react";
+import Login from "./Login";
```
Comment (colsen-neslo) [inline src/components/ManageLoginData.jsx]: Fix import ordering.

File: src/components/ManageLoginData.jsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+function ManageLoginData () {
```
Comment (colsen-neslo) [inline src/components/ManageLoginData.jsx]: Use arrow function.

File: src/components/ManageLoginData.jsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    const [login,setLogin] = useState({
+        username : "",
+        password : "",
+    });
+
+    function onLoginChange (event){
+        const updateDetails = {...login, [event.target.name]: event.target.value};
+        setLogin(updateDetails);
+    }
```
Comment (colsen-neslo) [inline src/components/ManageLoginData.jsx]:
```suggestion
    const onLoginChange = (newState) => {
        setLogin({...login, ...newState});
    }
```
Where the `newState` value is something like `{ username: 'exampleusername' }`.

File: src/components/login.css [hunk 10]
```diff
@@ -0,0 +1,26 @@
+div#login{
```
Comment (colsen-neslo) [inline src/components/login.css]:
```suggestion
#login {
```

File: src/components/login.css [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+    width: 100%;
+    border-radius: 10px;
+}
+
+div#login > form {
```
Comment (colsen-neslo) [inline src/components/login.css]:
```suggestion
#login > form {
```

File: src/index.css [hunk 12]
```diff
@@ -11,3 +11,23 @@ code {
...
+input{
+  width: 50%;
+  padding: 10px;
+  border-radius: 5px;
+  border: 1px solid #ccc;
+  margin-bottom: 10px;
+  box-sizing: border-box;
+}
+
+button{
```
Comment (colsen-neslo) [inline src/index.css]: Please make sure you format all your files before you commit your changes.
```suggestion
button {
```

File: src/index.js [hunk 13]
```diff
@@ -1,17 +1,15 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import App from './components/App';
+import {BrowserRouter} from "react-router-dom";
```
Comment (colsen-neslo) [inline src/index.js]: Fix import ordering.

Comment (colsen-neslo) [reply src/index.js]: This should also be a JSX file type.

**Review by colsen-neslo — CHANGES_REQUESTED** [review]
Please make sure you remove your `.idea` files.

File: package.json [hunk 14]
```diff
@@ -8,6 +8,7 @@
     "@testing-library/user-event": "^13.5.0",
```
Comment (noydb) [inline package.json]: these `@testing-library` dependencies I think can be moved to `devDependencies`

File: src/components/App.jsx (hunk 1 above)
Comment (noydb) [reply src/components/App.jsx]: and a whitespace please between the last import and the function/component declaration

File: src/components/App.jsx [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+return (
+        <div className="container-fluid">
```
Comment (noydb) [inline src/components/App.jsx]: This class does nothing

File: src/components/App.jsx [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+            <Header/>
+            <Routes>
```
Comment (noydb) [inline src/components/App.jsx:13]: please format your code:
```
<Routes>
    <Route ...etc
</Routes>
```

File: src/components/Login.jsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+                <form>
+                    <fieldset>
+                        <label htmlFor="userName">Username</label>
+                        <div className="field" id="fields">
+                            <input
+                                id="userName"
```
Comment (noydb) [inline src/components/Login.jsx]: this id is unnecessary. if you need to select this element, use `fieldset:first-of-type > div > input`

File: src/components/Login.jsx [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+                                value={props.login.username}
+                            />
+                        </div>
+                    </fieldset>
+
+                    <fieldset>
+                        <label htmlFor="password">password</label>
+                        <div className="field">
+                            <input
+                                id="password"
```
Comment (noydb) [inline src/components/Login.jsx:32]: unnecessary id

File: src/components/Login.jsx [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+    return (
+        <>
+            <h1>Login</h1>
+
+            <div id="login">
+
+                <form>
+                    <fieldset>
+                        <label htmlFor="userName">Username</label>
+                        <div className="field" id="fields">
```
Comment (noydb) [inline src/components/Login.jsx]: I think this div is unnecessary

File: src/components/Login.jsx [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+                            <input
+                                id="userName"
+                                onChange={props.onLoginChange}
+                                name="username"
+                                type="text"
+                                className="form-control"
```
Comment (noydb) [inline src/components/Login.jsx]: unnecessary class

File: src/components/Login.jsx [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+
+                    <fieldset>
+                        <label htmlFor="password">password</label>
+                        <div className="field">
+                            <input
+                                id="password"
+                                onChange={props.onLoginChange}
+                                name="password"
+                                type="text"
+                                className="form-control"
```
Comment (noydb) [inline src/components/Login.jsx]: unnecessary class

File: src/components/Login.jsx [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+                                value={props.login.username}
+                            />
+                        </div>
+                    </fieldset>
+
+                    <fieldset>
+                        <label htmlFor="password">password</label>
+                        <div className="field">
```
Comment (noydb) [inline src/components/Login.jsx]: remove this div

File: src/components/Login.jsx [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+                    <span>not a member ?&nbsp;</span>
+                    <NavLink to="/register">register</NavLink>
+                </div>
+            </div>
+        </>
+
+    )
+}
+
+export default Login;
```
Comment (noydb) [inline src/components/Login.jsx:38]: Please declare propTypes

File: src/components/ManageLoginData.jsx [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+
+    return (
+            <Login login={login} onLoginChange={onLoginChange} />
```
Comment (noydb) [inline src/components/ManageLoginData.jsx]: What is `login`? Passing in a property called `login` to a component called login is ambiguous. Please rename the `login` property to something more indicative of what the variable actually is

File: src/components/login.css [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+    border: 6px solid cyan;
+    border-radius: 12px;
+    padding: 4rem;
+    margin: 2rem 30rem;
+}
+
+h1{
```
Comment (noydb) [inline src/components/login.css]: format this file

File: src/components/login.css [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+#login button[type="submit"] {
+    width: 50%;
+    height: 50px;
+    border-radius: 14px;
+    background-color: chartreuse;
+}
+
+#login {
```
Comment (noydb) [inline src/components/login.css]: does the h1 need `text-align: center` if the parent element has `text-align: center`? If you do need to keep text-align for `#login`, then please group it with the `div#login` selector at the top

File: src/index.css [hunk 27]
```diff
@@ -11,3 +11,23 @@ code {
...
+  border: none;
+}
+
+input{
+  width: 50%;
+  padding: 10px;
+  border-radius: 5px;
+  border: 1px solid #ccc;
+  margin-bottom: 10px;
+  box-sizing: border-box;
```
Comment (noydb) [inline src/index.css]: I don't think this is needed

**Review by noydb — CHANGES_REQUESTED** [review]

File: src/components/Login.jsx (hunk 17 above)
Comment (noydb) [reply src/components/Login.jsx]: or actually I see you're using it for `htmlFor`, so you can keep it. but then please change the id to `username`

File: src/index.css (hunk 12 above)
Comment (TristanCarswell) [reply src/index.css]: Have we configured his IDE formatting settings yet? @colsen-neslo

Comment (colsen-neslo) [reply src/index.css]: This should be setup by default.

File: src/components/login.css (hunk 26 above)
Comment (Stanley-Maraba) [reply src/components/login.css]: it is outside login

File: src/components/ManageLoginData.jsx (hunk 9 above)
Comment (Stanley-Maraba) [reply src/components/ManageLoginData.jsx]: using it this way does not set the key so when trying to input something it will not allow please advice

Comment (colsen-neslo) [reply src/components/ManageLoginData.jsx]: So then change the way you are handling the onChange function of the input to produce this.

File: src/components/App.jsx [hunk 28]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import {Route, Router, Routes} from "react-router-dom";
+import Header from './header/Header'
+import ManageLoginData from './ManageLoginData';
+import HomePage from './HomePage';
+import ManageRegisterData from "./ManageRegisterData";
+import PageNotFound from './PageNotFound'
+const App = () => {
```
Comment (noydb) [inline src/components/App.jsx:9]: whitespace between function/component declaration and the last import line please

File: src/components/Login.jsx [hunk 29]
```diff
@@ -0,0 +1,54 @@
...
+                <div>
+                    <span>not a member ?&nbsp;</span>
+                    <NavLink to="/register">register</NavLink>
+                </div>
+            </section>
+        </>
+    )
+}
+
+Login.propTypes = {
```
Comment (noydb) [inline src/components/Login.jsx]: declare default props for these properties, due to the fact that they are not required

File: src/components/Login.jsx [hunk 30]
```diff
@@ -0,0 +1,54 @@
...
+                            <input
+                                id="password"
+                                onChange={(event) => props.onLoginChange({password: event.target.value})}
+                                name="password"
+                                type="text"
+                                value={props.login.password}
+                            />
+                        </div>
+                    </fieldset>
+                    <button className="btn btn-primary" type="submit">LOGIN</button>
```
Comment (noydb) [inline src/components/Login.jsx]: are the classes on this button needed?

File: src/components/login.css [hunk 31]
```diff
@@ -0,0 +1,26 @@
...
+    width: 100%;
+    border-radius: 10px;
+}
+
+login > form {
```
Comment (noydb) [inline src/components/login.css]: I don't think `login` is going to match/select anything

File: src/components/login.css [hunk 32]
```diff
@@ -0,0 +1,26 @@
...
+
+#login button[type="submit"] {
+    width: 50%;
+    height: 50px;
+    border-radius: 14px;
+    background-color: chartreuse;
+}
+
+#login {
+    text-align: center;
```
Comment (noydb) [inline src/components/login.css]: move text align center to the top rule: `#login {`

File: src/index.css [hunk 33]
```diff
@@ -11,3 +11,23 @@ code {
...
+  border: none;
+}
+
+input {
+  width: 50%;
+  padding: 10px;
+  border-radius: 5px;
+  border: 1px solid #ccc;
+  margin-bottom: 10px;
+    box-sizing: border-box;
```
Comment (noydb) [inline src/index.css]: do you need this?

**Review by noydb — CHANGES_REQUESTED** [review]

File: src/components/Login.jsx [hunk 34]
```diff
@@ -0,0 +1,54 @@
...
+                                name="username"
+                                type="text"
+                                value={props.login.username}
+                            />
+                        </div>
+                    </fieldset>
+
+                    <fieldset>
+                        <label htmlFor="password">password</label>
+                        <div>
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: Is this div required?

File: src/components/Login.jsx [hunk 35]
```diff
@@ -0,0 +1,54 @@
...
+                            <input
+                                id="username"
+                                onChange={(event) => props.onLoginChange({username: event.target.value})}
+                                name="username"
+                                type="text"
+                                value={props.login.username}
+                            />
+                        </div>
+                    </fieldset>
+
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: Remove unnecessary blank line.

File: src/components/Login.jsx [hunk 36]
```diff
@@ -0,0 +1,54 @@
...
+                    <NavLink to="/register">register</NavLink>
+                </div>
+            </section>
+        </>
+    )
+}
+
+Login.propTypes = {
+    username : PropTypes.string,
+    password : PropTypes.string
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: These prop types do not match the properties you are providing to your component, you are providing `login` and `loginChange` as your properties. Please make sure you define the prop types to reflect this.

File: src/components/ManageLoginData.jsx [hunk 37]
```diff
@@ -0,0 +1,18 @@
+import React,{useState} from "react";
+import Login from "./Login";
```
Comment (colsen-neslo) [inline src/components/ManageLoginData.jsx]: Fix import ordering based on early comments and discussion around import ordering, make sure your imports are clearly separated by a blank line.

File: src/components/ManageLoginData.jsx [hunk 38]
```diff
@@ -0,0 +1,18 @@
...
+
+const ManageLoginData = () => {
+    const [loginDetails,setLoginDetails] = useState({
+        username : "",
+        password : "",
+    });
```
Comment (colsen-neslo) [inline src/components/ManageLoginData.jsx]: Format your files.
```suggestion
    const [loginDetails, setLoginDetails] = useState({
        username : "",
        password : "",
    });
```

File: src/index.jsx [hunk 39]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import {BrowserRouter} from "react-router-dom";
+import App from './components/App';
+import './index.css';
```
Comment (colsen-neslo) [inline src/index.jsx]: Fix imports.

**Review by colsen-neslo — CHANGES_REQUESTED** [review]
Please remove your .idea files

File: src/components/Login.jsx (hunk 36 above)
Comment (noydb) [reply src/components/Login.jsx]: I asked in a previous comment for `login` prop to be changed as well. I find it ambiguous passing a `login` property to a component called `Login`

File: src/components/ManageLoginData.jsx (hunk 24 above)
Comment (noydb) [reply src/components/ManageLoginData.jsx]: @Stanley-Maraba you missed this comment. Please be careful when resolving comments yourself

File: src/components/Login.jsx (hunk 30 above)
Comment (Stanley-Maraba) [reply src/components/Login.jsx]: used it at first to get basic styling, resolved now

File: src/components/Login.jsx [hunk 40]
```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import PropTypes, { string } from "prop-types";
+import { NavLink } from "react-router-dom";
+
+import "./login.css";
+
+const Login = (props) => {
+    console.log(props.loginDetails.username)
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: Remove this

File: src/components/Login.jsx (hunk 34 above)
Comment (colsen-neslo) [reply src/components/Login.jsx]: ?

File: src/components/Login.jsx [hunk 41]
```diff
@@ -0,0 +1,61 @@
...
+            </section>
+        </>
+    )
+}
+Login.defaultProps = {
+   password : "",
+    username : ""
+}
+
+Login.propTypes = {
```
Comment (colsen-neslo) [inline src/components/Login.jsx]: Add your onchange function to the prop types too.

File: src/components/Login.jsx (hunk 34 above)
Comment (Stanley-Maraba) [reply src/components/Login.jsx]: yes

**Review by colsen-neslo — CHANGES_REQUESTED** [review]
@Stanley-Maraba please remove the .idea files from this pull request.

File: src/components/Login.jsx [hunk 42]
```diff
@@ -46,9 +47,15 @@ const Login = (props) => {
         </>
     )
 }
+Login.defaultProps = {
```
Comment (noydb) [inline src/components/Login.jsx]: whitespace above here please

File: src/components/Login.jsx (hunk 34 above)
Comment (noydb) [reply src/components/Login.jsx]: Why is it required?

File: src/components/Login.jsx [hunk 43]
```diff
@@ -0,0 +1,61 @@
...
+    console.log(props.loginDetails.username)
+
+    return (
+        <>
+            <h1>Login</h1>
+            <section id="login">
+                <form>
+                    <fieldset>
+                        <label htmlFor="username">Username</label>
+                        <div id="fields">
```
Comment (noydb) [inline src/components/Login.jsx]: remove this div

File: src/components/Login.jsx [hunk 44]
```diff
@@ -47,15 +42,18 @@ const Login = (props) => {
         </>
     )
 }
+
 Login.defaultProps = {
-   password : "",
-    username : ""
+    password : "",
+    username : "",
```
Comment (colsen-neslo) [inline src/components/Login.jsx]:
```suggestion
    loginDetails: {
        password : "",
        username : ""
    }
```

**Review by colsen-neslo — CHANGES_REQUESTED** [review]

File: src/components/Login.jsx [hunk 45]
```diff
@@ -0,0 +1,59 @@
...
+import { NavLink } from "react-router-dom";
+
+import "./login.css";
+
+const Login = (props) => {
+
+    return (
+        <>
+            <h1>Login</h1>
+            <section id="login">
```
Comment (JOlsen-Neslo) [inline src/components/Login.jsx:22]: We generally avoid applying IDs to elements unless necessary. Also, rather style your markup using class and element selectors than styling by ID

File: src/components/Login.jsx [hunk 46]
```diff
@@ -0,0 +1,59 @@
...
+                    <label htmlFor="username">Username</label>
+                    <fieldset>
+                            <input
+                                id="username"
+                                onChange={(event) => props.onLoginChange({username: event.target.value})}
+                                name="username"
+                                type="text"
+                                value={props.loginDetails.username}
+                            />
+                    </fieldset>
```
Comment (JOlsen-Neslo) [inline src/components/Login.jsx]: This sort of logic gets repeated often so rather create a reusable component

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

File: src/components/Login.jsx (hunk 45 above)
Comment (noydb) [reply src/components/Login.jsx:22]: I think id should be used here because class implies multiple elements using it. There is only one login section in this component. I think it's perfectly valid to select via ID in a stylesheet. I actually think we need to move more away from classes.

File: src/components/Login.jsx [hunk 47]
```diff
@@ -5,36 +5,31 @@ import { NavLink } from "react-router-dom";
...
                             <input
                                 id="username"
                                 onChange={(event) => props.onLoginChange({username: event.target.value})}
                                 name="username"
                                 type="text"
                                 value={props.loginDetails.username}
                             />
-                        </div>
                     </fieldset>
+                    <label htmlFor="password">password</label>
```
Comment (noydb) [inline src/components/Login.jsx]: why did you move this out of the fieldset?

File: src/components/Login.jsx (hunk 45 above)
Comment (noydb) [reply src/components/Login.jsx:22]: I would even suggest pure HTML selection, like `:first-of-child` etc. But I think we need the `login` id to prevent leaking styles right?

File: src/components/Login.jsx (hunk 47 above)
Comment (noydb) [reply src/components/Login.jsx]: If it's because label displays inline with the input, then add `display: block` to the label

File: src/components/Login.jsx [hunk 48]
```diff
@@ -0,0 +1,61 @@
...
+    loginDetails : {
+        password : "",
+        username : ""
+    },
+    onChange : () => {}
+}
+
+Login.propTypes = {
+    loginDetails : PropTypes.shape ({
+    username : PropTypes.string,
```
Comment (noydb) [inline src/components/Login.jsx]: indent these fields, please constantly format your files

File: src/components/ManageLoginData.jsx [hunk 49]
```diff
@@ -0,0 +1,19 @@
+import React,{ useState } from "react";
```
Comment (noydb) [inline src/components/ManageLoginData.jsx]: Please update all your files to use single quotes. Double quotes isn't wrong, but the convention at neslo is single quotes

File: src/components/Input.jsx [hunk 50]
```diff
@@ -9,30 +9,23 @@ const Input = (props) => {
...
+    inputVal: '',
 };
 
 Input.propTypes = {
-    loginDetails: PropTypes.shape({
-        username: PropTypes.string,
-        password: PropTypes.string
-    }),
+    inputVal: PropTypes.string,
     onChange: PropTypes.func.isRequired
```
Comment (colsen-neslo) [inline src/components/Input.jsx]: Missing the id prop.

File: src/components/Input.jsx [hunk 51]
```diff
@@ -9,30 +9,23 @@ const Input = (props) => {
                 <label htmlFor={props.id}>{props.id}</label>
```
Comment (colsen-neslo) [inline src/components/Input.jsx]: Rather use a label prop.

File: src/components/Input.jsx [hunk 52]
```diff
@@ -9,30 +9,23 @@ const Input = (props) => {
...
                 <input
                     id={props.id}
-                    onChange={(event) => props.onLoginChanges({[props.id]: event.target.value})}
+                    onChange={(event) => props.onChange({[props.id]: event.target.value})}
                     name={props.id}
                     type="text"
```
Comment (colsen-neslo) [inline src/components/Input.jsx]: Rather provide a type property.

File: src/components/Input.jsx (hunk 50 above)
Comment (colsen-neslo) [reply src/components/Input.jsx]: You will need to add the label and type props once you use them

File: src/components/Login.jsx [hunk 53]
```diff
@@ -12,8 +12,8 @@ const Login = (props) => {
             <h1>Login</h1>
             <section id="login">
                 <form>
-                    <Input id={'username'} onLoginChanges={props.onLoginChange} loginDetails={props.loginDetails}/>
-                    <Input id={'password'} onLoginChanges={props.onLoginChange} loginDetails={props.loginDetails}/>
+                    <Input id={'username'} onChange={props.onLoginChange} inputVal={props.loginDetails.username}/>
+                    <Input id={'password'} onChange={props.onLoginChange} inputVal={props.loginDetails.password}/>
```
Comment (colsen-neslo) [inline src/components/Login.jsx]:
```suggestion
                    <Input id="username" onChange={props.onLoginChange} inputVal={props.loginDetails.username}/>
                    <Input id="password" onChange={props.onLoginChange} inputVal={props.loginDetails.password}/>
```

Comment (colsen-neslo) [reply src/components/Login.jsx]: You can also handle the login form state change in this component instead of passing in another property to this component. You can then pass in a prop that handles the submission on the login button click. That function can consume the form state managed on this component level.

**Review by colsen-neslo — CHANGES_REQUESTED** [review]

File: src/components/Input.jsx [hunk 54]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Input = (props) => {
+
+    return (
+        <>
```
Comment (noydb) [inline src/components/Input.jsx]: you only need a `React.Fragment` when you have two parent elements in a React component. In this instance, you only have one element at the parent level (`fieldset`), so you can just render `fieldset` and remove the `<></>`

If you had this:
```
return (
    <div></div>
    <div></div>
);
```
It won't compile

to fix it, you'd do this: 
```
return (
    <>
        <div></div>
        <div></div>
    </>
);
```

**Review by noydb — CHANGES_REQUESTED** [review]

File: src/components/App.jsx [hunk 55]
```diff
@@ -0,0 +1,23 @@
...
+
+const App = () => {
+return (
+    <>
+            <Header/>
+            <Routes>
+                <Route path="/" element={<HomePage/>}/>
+                <Route path="/login" element={<ManageLoginData/>}/>
+                <Route path="/register" element={<ManageRegisterData/>}/>
+                {/*<Route element={<PageNotFound/>}/>*/}
```
Comment (JOlsen-Neslo) [inline src/components/App.jsx]: Please remove commented code

File: src/components/HomePage.jsx [hunk 56]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+const HomePage = () => {
+
+    return (
+
+        <h1>This is my home page</h1>
```
Comment (JOlsen-Neslo) [inline src/components/HomePage.jsx]: Please format this file

File: src/components/Input.jsx [hunk 57]
```diff
@@ -0,0 +1,34 @@
...
+                    value={props.inputVal}
+                />
+            </fieldset>
+        </>
+
+    );
+};
+
+Input.defaultProps = {
+    inputVal: '',
```
Comment (JOlsen-Neslo) [inline src/components/Input.jsx]: why `inputVal` and not `value`? Renaming this property is redundant

File: src/components/Login.jsx [hunk 58]
```diff
@@ -0,0 +1,38 @@
...
+        setLoginDetails({...loginDetails, ...newState});
+    };
+
+    return (
+        <>
+            <h1>Login</h1>
+            <section id="login">
+                <form>
+                    <Input id="username" onChange={onLoginChange} inputVal={loginDetails.username} label="username"
+                           type="text"/>
```
Comment (JOlsen-Neslo) [inline src/components/Login.jsx]: if you are defaulting type to `text` - no need to provide it here

File: src/components/Login.jsx [hunk 59]
```diff
@@ -0,0 +1,38 @@
...
+                    <button type="submit" onClick={props.submit}>LOGIN</button>
+                </form>
+                <div>
+                    <span>not a member ?&nbsp;</span>
+                    <NavLink to="/register">register</NavLink>
+                </div>
+            </section>
+        </>
+    );
+};
```
Comment (JOlsen-Neslo) [inline src/components/Login.jsx:36]:
```suggestion
};

```

File: src/components/ManageLoginData.jsx [hunk 60]
```diff
@@ -0,0 +1,11 @@
...
+
+import Login from './Login';
+
+const ManageLoginData = () => {
+
+    return (
+        <Login/>
+    );
+};
+export default ManageLoginData;
```
Comment (JOlsen-Neslo) [inline src/components/ManageLoginData.jsx]: this can be clean up

```suggestion
const ManageLoginData = () => (
     <Login/>
);

export default ManageLoginData;
```

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

File: src/components/Input.jsx [hunk 61]
```diff
@@ -0,0 +1,33 @@
...
+
+    return (
+            <fieldset>
+                <label htmlFor={props.id}>{props.label}</label>
+                <input
+                    id={props.id}
+                    onChange={(event) => props.onChange({[props.id]: event.target.value})}
+                    name={props.id}
+                    type={props.type}
+                    value={props.inputVal}
```
Comment (TristanCarswell) [inline src/components/Input.jsx]: Should this not be props.value? Since in your prop types below it is defined as value

Comment (Stanley-Maraba) [reply src/components/Input.jsx]: should be inputVal made mistake on the prop types

File: src/components/App.jsx [hunk 62]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import {Route, Router, Routes} from "react-router-dom";
```
Comment (noydb) [inline src/components/App.jsx]: single quotes here please

File: src/components/Input.jsx [hunk 63]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Input = (props) => {
+
+    return (
+            <fieldset>
+                <label htmlFor={props.id}>{props.label}</label>
+                <input
+                    id={props.id}
```
Comment (noydb) [inline src/components/Input.jsx]: `props.id` is being repeated three times, please assign a variable to the value or destructure `props`

File: src/components/login.css [hunk 64]
```diff
@@ -0,0 +1,27 @@
...
+
+#login button[type="submit"] {
+    width: 50%;
+    height: 50px;
+    border-radius: 14px;
+    background-color: chartreuse;
+}
+
+label {
+    display: block
```
Comment (noydb) [inline src/components/login.css]: missing semi-colon

**Review by TristanCarswell — APPROVED** [review]

File: src/components/Input.jsx [hunk 65]
```diff
@@ -0,0 +1,34 @@
...
+            <label htmlFor={inputId}>{props.label}</label>
+            <input
+                id={inputId}
+                onChange={(event) => props.onChange({[inputId]: event.target.value})}
+                name={inputId}
+                type={props.type}
+                value={props.value}
+            />
+        </fieldset>
+
```
Comment (noydb) [inline src/components/Input.jsx]: unnecessary blank line

**Review by noydb — APPROVED** [review]
